base/special/exp.jl

```diff
@@ -1,138 +1,194 @@
 
+@inline function table_unpack(ind)
+    j = @inbounds J_TABLE[ind]
+    jU = reinterpret(Float64, JU_CONST | (j&JU_MASK))
+    jL = reinterpret(Float64, JL_CONST | (j>>8))
+    return jU, jL
+end
+
+
+"""
+    exp10(x)
+Compute ``10^x``.
 # Examples
 ```jldoctest
-julia> exp(1.0)
-2.718281828459045
+julia> exp10(2)
+100.0
+julia> exp10(0.2)
+1.5848931924611136
 ```
 """
-exp(x::Real) = exp(float(x))
-function exp(x::T) where T<:Union{Float32,Float64}
-    xa = reinterpret(Unsigned, x) & ~sign_mask(T)
-    xsb = signbit(x)
-
-    # filter out non-finite arguments
-    if xa > reinterpret(Unsigned, MAX_EXP(T))
-        if xa >= exponent_mask(T)
-            xa & significand_mask(T) != 0 && return T(NaN)
-            return xsb ? T(0.0) : T(Inf) # exp(+-Inf)
-        end
-        x > MAX_EXP(T) && return T(Inf)
-        x < MIN_EXP(T) && return T(0.0)
-    end
-    # This implementation gives 2.7182818284590455 for exp(1.0) when T ==
-    # Float64, which is well within the allowable error; however,
-    # 2.718281828459045 is closer to the true value so we prefer that answer,
-    # given that 1.0 is such an important argument value.
-    if x == T(1.0) && T == Float64
-        return 2.718281828459045235360
-    end
-    # compute approximation
-    if xa > reinterpret(Unsigned, T(0.5)*T(LN2)) # |x| > 0.5 log(2)
-        # argument reduction
-        if xa < reinterpret(Unsigned, T(1.5)*T(LN2)) # |x| < 1.5 log(2)
-            if xsb
-                k = -1
-                hi = x + LN2U(T)
-                lo = -LN2L(T)
-            else
-                k = 1
-                hi = x - LN2U(T)
-                lo = LN2L(T)
+myexp10(x)
```

Comment:
```suggestion
exp10(x)
```

But I also think this doc statement is redundant with the one for `exp10(x::AbstractFloat)` which is already in `base/math.jl`.

Reply:
I might have just fixed it.

Reply:
I think I edited about the same time — it might need `exp(x::Real)` in the `@doc` block versus just a bare `exp` since there is a matrix-specific doc string for `exp(::AbstractMatrix)` already... Not really sure about this since I can't trivially check it. (`Revise.track(Base)` doesn't seem to delete docstrings for removed doc sources...)

Reply:
Also, I think you do have to have the `@doc "..." exp(x::Real)` block _after_ the loop that defines the functions. If it's placed before the method is defined, the doc isn't properly "attached" to the method (so I think that's why the latest round of CI is failing again.)

Reply:
I'm trying this now. We'll see if it works.